doc/contributing.rst

```diff
@@ -15,4 +20,116 @@ Contributions to RSMTool are very welcome. You can use the instructions below to
+        |   field in the configuration file.                                         |
+        +----------------------------------------------------------------------------+
+        | Writing test(s) for RSMEval                                                |
+        |                                                                            |
+        | * Same arguments as RSMTool except the ``skll`` keyword argument is not    |
+        |   applicable.                                                              |
+        +----------------------------------------------------------------------------+
+        | Writing test(s) for RSMPredict                                             |
+        |                                                                            |
+        | * Only positional argument is the name of the test directory you created.  |
+        |                                                                            |
+        | * Use ``excluded=True`` if you want to check the excluded responses file   |
+        |   as part of the test.                                                     |
+        |                                                                            |
+        | * Set ``file_format="tsv"`` (or ``"xlsx"``) if you specified the same      |
+        |   field in the configuration file.                                         |
+        +----------------------------------------------------------------------------+
+        | Writing test(s) for RSMCompare                                             |
+        |                                                                            |
+        | * First positional argument is the name of the test directory you created. |
+        |                                                                            |
+        | * Second positional argument is the comparison ID from the JSON            |
+        |   configuration file.                                                      |
+        +----------------------------------------------------------------------------+
+        | Writing test(s) for RSMSummarize                                           |
+        |                                                                            |
+        | * Only positional argument is the name of the test directory you created.  |
+        |                                                                            |
+        | * Set ``file_format="tsv"`` (or ``"xlsx"``) if you specified the same      |
+        |   field in the configuration file.                                         |
+        +----------------------------------------------------------------------------+
+
+Once you have added all new functional tests, you should run ``nosetests --noglogcapture`` to run all the tests. Obviously, the newly added tests will fail since you have not yet generated the expected output for that test. To do this, you should now run the following:
```

Comment:
Somewhere here should be a step where one reviews the outputs and makes sure they are as expected before adding them to the test.

Reply:
Also, I would add a recommendation to commit changes before updating test files. In this case if updating the files went terribly wrong you can just stash them and start from the last commit.

Reply:
Yup, good ideas. Will add those steps.